### ISSUE TITLE
ErrorsController - Directly Calling "Errors/Other/400" Returns Default 500 Instead of 400

### ISSUE THREAD

**Issue by iokevins** [iaingalloway/ASP.NET-HttpErrors-Example#2]
Currently, this works (returns 403):
http://httperrors-example.azurewebsites.net/Error/Other/403

But not this (returns default, 500):
http://httperrors-example.azurewebsites.net/Errors/Other/400

It seems the catchall route, "{controller}/{action}/{id}", handles the latter request, passing parameter "id=400" to ErrorsController action Other, which sets optional parameter code=null. 

One fix: 1) change ErrorsController parameter name for action Other(...), from code, to id; and 2) change route "Error/{action}/{code}" to "Error/{action}/{id}".

After doing so, both URLs work as expected. 

Note: noting only for future reference, as others may wish to directly call the Errors controller rather than handling via IIS.

Thank you,

Kevin

**Comment by iaingalloway** [comment]
I'm not sure I understand what your expected behaviour is here. The error controller isn't intended to be invoked directly via the default route. It *replaces* the original response as per the settings in the web.config.

The example links on the index don't refer to the error controller by its route, but instead simulate actual error conditions in your application (by raising an exception for 500, setting a status code for 403, and not existing at all for 404).

If for some reason you want to be able to call the the Other action on the Errors controller like "Errors/Other/400" then sure you'll need to either rename the parameter, or alter the routes - but that's not really the purpose of this demo.

As a tangent, if for some reason you *do* call the error controller directly, then the error replacement configured in the web.config *still happens*. For example, if you e.g. call http://httperrors-example.azurewebsites.net/Errors/Other/?code=403 - the 403 status code set by the controller gets (correctly) intercepted by IIS and the /Error/Forbidden action is rendered *instead*.

**Comment by iokevins** [comment]
Hi Iain,

Thank you; after reading your response, I find myself in complete agreement and with an even better understanding of how it works--e.g., /Errors/Other/?code=403 > /Error/Forbidden > /Errors/Forbidden.

Looking back, my issue does seem worded rather confusingly--it arises after a day or two of wrestling with two implementation issues:

1. In my case, I had to change web.config path, from path="/Error/Forbidden" to path="/MyAppFolder/Error/Forbidden" (that is, rooted to web site, not web app subfolder)...otherwise I got empty/blank screen content with 403 forbidden or 404 not found errors

2. The above confusion (on my part) about directly accessing controllers 

Now I have resolved those issues and wanted to share what worked for me, so others might also use this solution successfully.

I'm grateful. Thank you again.

Kevin
